pyro/poutine/trace_struct.py

```diff
@@ -306,10 +320,91 @@ def pack_tensors(self, plate_to_symbol=None):
                     packed["unscaled_log_prob"] = pack(site["unscaled_log_prob"], dim_to_symbol)
             except ValueError:
                 _, exc_value, traceback = sys.exc_info()
+                shapes = self.print_shapes(last_site=site["name"])
                 six.reraise(ValueError,
-                            ValueError("Error while packing tensors at site '{}':\n  {}"
-                                       .format(site["name"], exc_value)),
+                            ValueError("Error while packing tensors at site '{}':\n  {}\n{}"
+                                       .format(site["name"], exc_value, shapes)),
                             traceback)
 
-        self.plate_to_symbol = plate_to_symbol
-        self.symbol_to_dim = symbol_to_dim
+    def print_shapes(self, title='Trace Shapes:', last_site=None):
```

Comment:
nit: Would suggest calling this method `repr_shapes` or something like that, which suggests that it would return a string representation.